src/System.Web.OData/OData/EnableQueryAttribute.cs

```diff
@@ -162,6 +162,23 @@ public int PageSize
         }
 
         /// <summary>
+        /// Honor $filter inside $expand of non-collection navigation property.
+        /// The expanded property is only populated when the filter evaluates to true.
+        /// This setting is false by default.
+        /// </summary>
+        public bool HandleReferenceNavigationPropertyExpandFilter
```

Comment:
Is the idea of the setting just to maintain back-compat with the existing behavior, as opposed to wanting to turn it on or off?

Reply:
back-compat. By making this opt-in, this change does not change existing consumers and easier to accept :)